### PR TITLE
[vpc-peering] Implement cross-account peering

### PR DESCRIPTION
**Pull request by osterman**
## what
* Implement cross-account VPC peering

## why
* Support use-cases where one VPC needs to access another e.g. access the RDS instance in the other VPC

### CODE REVIEW

File: aws/vpc-peering/main.tf [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+variable "requester_region" {
+  default = "us-west-2"
+}
+
+variable "requester_account" {
+}
+
+variable "requester_vpc_tags" {
+  type = "map"
+  default = {}
```
Comment (aknysh) [inline aws/vpc-peering/main.tf]: `terraform fmt`

File: aws/vpc-peering/main.tf [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+variable "accepter_region" {
+  default = "data"
```
Comment (aknysh) [inline aws/vpc-peering/main.tf]:
```suggestion
  default = "us-west-2"
```

File: aws/vpc-peering/variables.tf [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+variable "name" {
+  type        = "string"
+  description = "Application or solution name (e.g. `app`)"
+  default     = "vpc-peering"
+}
+
+variable "requester_account" {
+  description = "Account name of the requester (e.g. `prod` or `staging`). USed to look up the role ARN from SSM."
```
Comment (aknysh) [inline aws/vpc-peering/variables.tf]:
```suggestion
  description = "Account name of the requester (e.g. `prod` or `staging`). Used to look up the role ARN from SSM"
```

File: aws/vpc-peering/variables.tf [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  description = "Tags to filter for the requester's VPC"
+  default     = {}
+}
+
+variable "accepter_region" {
+  decription = "Region of the accepter's VPC"
+}
+
+variable "accepter_account" {
+  description = "Account name of the accepter (e.g. `prod` or `staging`). Used to look up the role ARN from SSM."
```
Comment (aknysh) [inline aws/vpc-peering/variables.tf]:
```suggestion
  description = "Account name of the accepter (e.g. `prod` or `staging`). Used to look up the role ARN from SSM"
```

File: aws/vpc-peering/main.tf [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+# Fetch the OrganizationAccountAccessRole ARNs from SSM
+module "requester_role_arns" {
+  enabled        = "${var.enabled}"
+  source         = "git::https://github.com/cloudposse/terraform-aws-ssm-parameter-store?ref=tags/0.1.5"
+  parameter_read = ["/${var.namespace}/${var.requester_account}/organization_account_access_role"]
+}
+
+locals {
+  requester_vpc_tags = "${var.requester_vpc_tags}"
```
Comment (aknysh) [inline aws/vpc-peering/main.tf]: looks like these two locals are redundant

File: aws/vpc-peering/main.tf [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+# Fetch the OrganizationAccountAccessRole ARNs from SSM
+module "accepter_role_arns" {
+  enabled        = "${var.enabled}"
+  source         = "git::https://github.com/cloudposse/terraform-aws-ssm-parameter-store?ref=tags/0.1.5"
+  parameter_read = ["/${var.namespace}/${var.accepter_account}/organization_account_access_role"]
+}
+
+locals {
+  accepter_vpc_tags = "${var.accepter_vpc_tags}"
+  accepter_region   = "${var.accepter_region}"
```
Comment (aknysh) [inline aws/vpc-peering/main.tf]: looks like these two locals are redundant

Comment (osterman) [reply aws/vpc-peering/main.tf]: Just for symmetry ....

**Pull request merged**